internal/provider/utils/utils.go

```diff
@@ -9,3 +14,38 @@ func SetStringValue(input string) types.String {
 	}
 	return types.StringNull()
 }
+
+// Helper function to map the properties response from API to Terraform state
+func HandlePropertiesMapping(ctx context.Context, propertiesAttr types.Map, apiProperties map[string]interface{}) types.Map {
+	// Map API properties to Terraform format
+	properties := map[string]attr.Value{}
+	for key, value := range apiProperties {
+		if value != nil {
+			properties[key] = types.StringValue(value.(string))
+		}
+	}
+
+	// Retrieve the current Terraform state for `properties`
+	var currentPropertiesState map[string]string
+	diags := propertiesAttr.ElementsAs(ctx, &currentPropertiesState, false)
+	if diags.HasError() {
+		tflog.Error(ctx, "Error reading current properties state", map[string]interface{}{"errors": diags.Errors()})
+	}
+
+	// API response is empty
+	if len(properties) == 0 {
+		// No properties in config
+		if currentPropertiesState == nil {
+			return types.MapNull(types.StringType)
+		}
+		// Empty properties in config
```

Comment:
```suggestion
		//Otherwise, there are empty properties in the config, return an empty map
```

Reply:
No, I want to explicitly comment that properties were set as {} in the config. This is important distinction from properties=null. I prefer my version.